e2e/test/scenarios/dashboard-cards/click-behavior.cy.spec.js

```diff
@@ -0,0 +1,711 @@
+};
+
+const QUESTION_LINE_CHART = {
+  name: "Line chart",
+  display: "line",
+  query: {
+    aggregation: [["count"]],
+    breakout: [["field", ORDERS.CREATED_AT, { "temporal-unit": "month" }]],
+    "source-table": ORDERS_ID,
+    limit: 5,
+  },
+};
+
+const OBJECT_DETAIL_CHART = {
+  display: "object",
+  query: {
+    "source-table": ORDERS_ID,
+  },
+};
+
+const TARGET_QUESTION = {
+  ...QUESTION_LINE_CHART,
+  name: "Target question",
+};
+
+const DASHBOARD_FILTER_TEXT = createMockActionParameter({
+  id: "1",
+  name: "Text filter",
+  slug: "filter-text",
+  type: "string/=",
+  sectionId: "string",
+});
+
+const DASHBOARD_FILTER_TIME = createMockActionParameter({
+  id: "2",
+  name: "Time filter",
+  slug: "filter-time",
+  type: "date/month-year",
+  sectionId: "date",
+});
+
+const QUERY_FILTER_CREATED_AT = [
+  "between",
+  ["field", ORDERS.CREATED_AT, null],
+  "2022-08-01",
+  "2022-08-31",
+];
+
+const QUERY_FILTER_QUANTITY = [
+  "=",
+  ["field", ORDERS.QUANTITY, null],
+  POINT_COUNT,
+];
+
+describe("scenarios > dashboard > dashboard cards > click behavior", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  describe("dashcards without click behavior", () => {
+    it("does not allow to set click behavior for virtual dashcards", () => {
+      const textCard = createTextCard({ size_y: 1 });
+      const headingCard = createHeadingCard();
+      const actionCard = createActionCard();
+      const linkCard = createLinkCard();
+      const cards = [textCard, headingCard, actionCard, linkCard];
+
+      cy.createDashboard().then(({ body: dashboard }) => {
+        updateDashboardCards({ dashboard_id: dashboard.id, cards });
+        visitDashboard(dashboard.id);
+      });
+
+      editDashboard();
+
+      cards.forEach((card, index) => {
+        const display = card.visualization_settings.virtual_card.display;
+        cy.log(`does not allow to set click behavior for "${display}" card`);
+
+        getDashboardCard(index).realHover().icon("click").should("not.exist");
+      });
+    });
+
+    it("does not allow to set click behavior for object detail dashcard", () => {
+      cy.createQuestionAndDashboard({
+        questionDetails: OBJECT_DETAIL_CHART,
+      }).then(({ body: card }) => {
+        visitDashboard(card.dashboard_id);
+      });
+
+      editDashboard();
+
+      getDashboardCard().realHover().icon("click").should("not.exist");
+    });
+  });
+
+  describe("line chart", () => {
+    const questionDetails = QUESTION_LINE_CHART;
+
+    it("should open drill-through menu as a default click-behavior", () => {
+      cy.createQuestionAndDashboard({ questionDetails }).then(
+        ({ body: card }) => {
+          visitDashboard(card.dashboard_id);
+        },
+      );
+
+      clickLineChartPoint();
+      assertDrillThroughMenuOpen();
+    });
+
+    it("does not allow setting dashboard as custom destination if user has no permissions to it", () => {
```

Comment:
nit: The no permission test case for dashboard comes first while no permission test case for questions comes last. This is also inconsistent with the test listed in [this comment](https://github.com/metabase/metabase/issues/31498#issuecomment-1771094663) (no permission comes last in that comment).

Reply:
df4564e5c9a2788e9343869174861b2bcf1bb590